src/utils.ts

```diff
@@ -1,6 +1,28 @@
 import { Result, ok, err } from './result'
 import { ResultAsync } from './result-async'
 
+/**
+ * Wraps a function with a try catch, creating a new function with the same
+ * arguments but returning `Ok` if successful, `Err` if the function throws
+ *
+ * @param fn function to wrap with ok on success or err on failure
+ * @param errorFn when an error is thrown, this will wrap the error result if provided
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function fromThrowable<Fn extends (...args: readonly any[]) => any, E>(
+  fn: Fn,
+  errorFn?: (e: unknown) => E,
```

Comment:
Question for you: If `fromThrowable` is called without a `errorFn` argument, then what does the tsc compiler infer `E` to be? I don't know for certain (as I haven't actually checked out the code), but I think this situation may require having conditional types. 

https://www.typescriptlang.org/docs/handbook/2/conditional-types.html

Reply:
If the errorFn is not provided, `E` will be `unknown`. Which is not amazing, but I don't think there is a way to infer the error type, so is probably the most accurate.
Another option would be to make `errorFn` required

Reply:
Having the error function as optional is preferable. Having an `unknown` error value by default is totally fine in my opinion. 

As long as `any` types are not leaked I'm happy!